openwisp_monitoring/device/static/monitoring/js/floorplan.js

```diff
@@ -0,0 +1,410 @@
+            name: node.device_name,
+            status: node.monitoring.status,
+            location: node.coordinates,
+            "Mac address": node.mac_address,
+          };
+        });
+        return data;
+      },
+
+      onReady() {
+        const map = this.leaflet;
+        maps[currentFloor] = map;
+        // remove default geo map tiles
+        map.eachLayer((layer) => layer._url && map.removeLayer(layer));
+        const img = new Image();
+        img.src = imageUrl;
+        let initialZoom;
+        img.onload = () => {
+          const aspectRatio = img.width / img.height;
+          const h = img.height;
+          const w = h*aspectRatio;
+          const zoom = map.getMaxZoom()-1;
+
+          // To make the image center in the map at (0,0) coordinates
+          const anchorLatLng = L.latLng(0, 0);
+          const anchorPoint = map.project(anchorLatLng, zoom);
+
+          // Calculate the bounds of the image, with respect to the anchor point (0, 0)
+          // Leaflet's pixel coordinates increase to the right and downwards
+          // Unlike cartesian system where y increases upwards
+          // So top-left will have negative y and bottom-right will have positive y
+          // Similarly left will have negative x and right will have positive x
+          const topLeft = L.point(anchorPoint.x - w / 2, anchorPoint.y - h / 2);
+          const bottomRight = L.point(anchorPoint.x + w / 2, anchorPoint.y + h / 2);
+
+          // Update node coordinates to fit the image overlay
+          // We get the node coordinates from the API in the format for L.CRS.Simple
+          // So the coordinates is in for cartesian system with origin at top left corner
+          // Rendering image in the third quadrant with topLeft as (0,0) and bottomRight as (w,-h)
+          // So we convert py to positive and then project the point to get the corresponding topLeft
+          // Then unproject the point to get the corresponding latlng on the map
+          this.data.nodes.forEach((node) => {
+            const px = Number(node.coordinates.lng);
+            const py = -Number(node.coordinates.lat);
+            const nodeProjected = L.point(topLeft.x + px, topLeft.y + py);
+            const nodeLatLng = map.unproject(nodeProjected, zoom);
+            node.properties = {
+              ...node.properties,
+              location: nodeLatLng,
+            };
+          });
+          // Update the map options with new node locations
+          const mapOptions = this.utils.generateMapOption(this.data, this);
+          this.echarts.setOption(mapOptions);
+
+          // Unproject the topLeft and bottomRight points to get northWest and southEast latlngs
+          const nw = map.unproject(topLeft, zoom);
+          const se = map.unproject(bottomRight, zoom);
+          const bnds = L.latLngBounds(nw, se);
+          L.imageOverlay(imageUrl, bnds).addTo(map);
+          map.fitBounds(bnds);
+          map.setMaxBounds(bnds);
```

Comment:
```suggestion
                    map.setMaxBounds(bnds.pad(0.3));
```

I have some devices which are situated the edge of the map. Without the padding for max bound, the pop-up was getting clipped. 

<img width="612" height="610" alt="Screenshot from 2025-08-25 20-47-13" src="https://github.com/user-attachments/assets/c1b86aff-605b-4ec7-be01-8dc0103ea206" />

Adding padding to the map fixes this issue for me.

Reply:
With padding 
<img width="620" height="606" alt="Screenshot from 2025-08-25 20-48-32" src="https://github.com/user-attachments/assets/44c4d7b2-39a7-4934-8dc0-7e389c7ff34f" />